lib/rester.rb

```diff
@@ -21,5 +21,32 @@ def connect(service, params={})
       adapter = Client::Adapters.connect(service, adapter_opts)
       Client.new(adapter, params)
     end
+
+    def consumer_name
+      @_consumer_name ||= defined?(Rails) ? Rails.application.class.parent_name
+        : "Consumer"
+    end
+
+    def consumer_name=(name)
+      @_consumer_name = name
+    end
+
+    def correlation_id
+      _correlation_ids[Thread.current.object_id] ||= SecureRandom.uuid
+    end
+
+    def correlation_id=(id)
+      if id.nil?
+        _correlation_ids.delete(Thread.current.object_id)
+      else
+        _correlation_ids[Thread.current.object_id] = id
+      end
+    end
+
+    private
+
+    def _correlation_ids
+      @_correlation_ids ||= ThreadSafe::Cache.new
+    end
```

Comment:
This isn't thread safe, since multiple threads may call this while `@_correlation_ids` is `nil`, so it needs to be initialized synchronously.

Could potentially just do:
```ruby
@_correlation_ids = ThreadSafe::Cache.new
def _correlation_ids
  @_correlation_ids
end
```